MovingBlocks/DestinationSol
# Module select UI

# Description
This pull request adds a module select screen, which allows the player to choose which modules to use when starting a new game.

![image](https://user-images.githubusercontent.com/24301287/177849847-c4d240e9-4a69-4cfb-8cd3-337460360f22.png)

# Testing
- Continue an existing game (to ensure that this still works).
- Select the `Play Game` item in the main menu, followed by the `New Game` option. A <kbd>Modules</kbd> button should be visible in the bottom-left corner.
- Select the `Modules` button and the modules screen should appear.
- Create a few new games to test the following things:
  - When the `warp` module (or any other module, as applicable) is de-activated, you should not be able to select warp ships. Warp features, such as wormholes, should not appear in-game.
  - When the game is resumed, it should still only contain the modules selected upon its creation. No new modules should have be running, even if added after the save was created.

# Known Issues
- Console commands persist event after their module is removed. This is because the console is part of the application context, rather than the game context, which isn't unloaded until the application exits.
- Modifications made to NUI screens by modules persist even after that module is removed.

# Notes
 - This depends on #647.
 - Feel free to comment on the UI designs.
